xarray/core/dataset.py

```diff
@@ -7293,7 +7293,7 @@ def from_dataframe(cls, dataframe: pd.DataFrame, sparse: bool = False) -> Self:
         arrays = []
         extension_arrays = []
         for k, v in dataframe.items():
-            if not is_extension_array_dtype(v) or isinstance(
+            if not is_allowed_extension_array(v) or isinstance(
                 v.array, UNSUPPORTED_EXTENSION_ARRAY_TYPES
```

Comment:
I didn't add `UNSUPPORTED_EXTENSION_ARRAY_TYPES` to the new `is_allowed_extension_array` function because we do allow them as backing arrays to `Index` object, I think.  Maybe should get a test?

Reply:
I think we need to account for internal duck array support i.e., that which allows preserving the `dtype` of extension array indices that are in this `UNSUPPORTED_EXTENSION_ARRAY_TYPES` whitelist.  See the note here:

https://github.com/pydata/xarray/blob/5ce69b2b993874bda88dd1670b125b964779eeb5/xarray/core/extension_array.py#L101-L103